stack/argparse-script-topic-type.rst

```diff
@@ -0,0 +1,120 @@
+
+For example, if the module’s namespace is ``lsst.example``, the full path for the file within the package repository is :file:`doc/lsst.example/scripts/myScript.py.rst`.
+
+.. _argparse-script-topic-autoprogram:
+
+The autoprogram directive
+=========================
+
+Everything in an argparse-based script topic is generated from a single autoprogram_ directive, which is provided by the `sphinxcontrib.autoprogram`_ Sphinx extension.
+autoprogram_ generates all the content that is described by the :doc:`script topic type <script-topic-type>`, including the program_ and option_ directives for cross-referencing.
+
+To use the autoprogram_ directive, your script needs to be set up in a particular fashion:
+
+- The script needs to be in the ``bin.src`` directory of a package, but the script file must defer its implementation to a module *inside* the package's Python modules.
+
+  For example, a script file :file:`bin.src/exampleScript.py` might be structured like this:
+
+  .. code-block:: python
+
+     #! /usr/bin/env python
+
+     from lsst.example.scripts.exampleScript import main
+
+
+     if __name__ == "__main__":
+         main()
+
+- The `argparse.ArgumentParser` instance must be generated by an argument-less function.
+  This is critical for letting the autoprogram_ directive get a copy of the `~argparse.ArgumentParser` instance to introspect the command-line interface:
+
+  .. code-block:: python
+
+     """Example script that is automatically documented.
+     """
+
+     from argparse import ArgumentParser
+
+
+     def build_argparser():
+         parser = ArgumentParser(
+            description=__doc__,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog='More information is available at https://pipelines.lsst.io.')
+         )
+         parser.add_argument(
+            'message'
+            help='Message to echo.'
+         )
+         return parser
+
+
+     def main():
+         args = build_argparser().parse_args()
+         print(args.message)
+
+  In this example, the ``build_argparser()`` function generates the `~argparse.ArgumentParser` instance.
+
+The argument to the autoprogram_ directive points to the function in your script's implementation that generates the `~argparse.ArgumentParser` instance.
+The argument is formatted as ``{module}:{function}()``, with a colon (``:``) separating the module from the function name.
+Given the above example, the autoprogram_ directive is written as:
+
+.. code-block:: rst
+
+   .. autoprogram:: lsst.example.scripts.exampleScript:build_argparser()
+      :prog: exampleScript.py
+      :groups:
+
+Set the ``:prog:`` option field to the name of the command-line executable.
+If you already set the ``prog`` argument in `argparse.ArgumentParser`, this field is not necessary.
+
+Also, the ``:groups:`` option field ensures that documentation is organized around argument groups (see `argparse.ArgumentParser.add_argument_group`), if they exist.
```

Comment:
option field --> optional field ?

Reply:
It's a directive option. I can just call it `:groups: option` if that sounds better to you.